src/core/models/highlightedvalue.js

```diff
@@ -135,6 +144,56 @@ export default class HighlightedValue {
     return highlightedValue;
   }
 
+  /**
+   * introduces highlighting to input data according to highlighting specifiers and returns a
+   * list of elements/strings
+   *
+   * @param {Object} val input object to apply highlighting to
+   * @param {Object} highlightedSubstrings highlighting specifiers to apply to input object
+   * @param {Function} transformFunction function to apply to strings in between highlighting markup
+   * @returns {Array<Element|string>} a list of elements/strings representing the val with
+   * highlighting applied. Example returned value :
+   *   [
+   *     'Save time & bank on your terms at over 1,800',
+   *     <strong>ATM</strong>,
+   *   ]
+   */
+  buildHighlightedElements (
```

Comment:
nit: if this is just an internal helper function, can the name begin with an underscore?

Reply:
updated!